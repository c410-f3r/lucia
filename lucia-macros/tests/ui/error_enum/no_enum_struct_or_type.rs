#[lucia_macros::pkg(api(Foo), data_format(json))]
mod pkg {
  #[pkg::req_data]
  fn foo() {}

  #[pkg::res_data]
  struct Res;
}

#[lucia_macros::pkg(api(Foo), data_format(json))]
mod pkg {
  #[pkg::req_data]
  struct Req;

  #[pkg::res_data]
  fn foo() {}
}

#[lucia_macros::pkg(api(Foo), data_format(json))]
mod pkg {
  #[pkg::params]
  fn foo() {}

  #[pkg::req_data]
  struct Req;

  #[pkg::res_data]
  struct Res;
}

fn main() {
}
