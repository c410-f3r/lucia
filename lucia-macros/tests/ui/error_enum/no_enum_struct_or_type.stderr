error: Invalid item. Expected enum, struct or type.
 --> tests/ui/error_enum/no_enum_struct_or_type.rs:3:10
  |
3 |   #[pkg::req_data]
  |          ^^^^^^^^

error: Invalid item. Expected enum, struct or type.
  --> tests/ui/error_enum/no_enum_struct_or_type.rs:15:10
   |
15 |   #[pkg::res_data]
   |          ^^^^^^^^

error: Invalid item. Expected enum, struct or type.
  --> tests/ui/error_enum/no_enum_struct_or_type.rs:21:10
   |
21 |   #[pkg::params]
   |          ^^^^^^
